torch/lib/c10d/test/ProcessGroupMPITest.cpp

```diff
@@ -36,13 +41,14 @@ void testAllreduce(int iter = 1000) {
 
   std::vector<c10::intrusive_ptr<::c10d::ProcessGroup::Work>> works;
   for (auto& tensors : allTensors) {
-    // Kick off work
     c10::intrusive_ptr<::c10d::ProcessGroup::Work> work =
         pg->allreduce(tensors);
     works.push_back(std::move(work));
   }
 
-  waitWork(pg, works);
+  std::vector<std::vector<at::Tensor>> outputTensors;
+
+  waitWork(pg, works, &outputTensors);
```

Comment:
Looks OK to me. Just curious, what's the benefit of adding the `need_output` flag, instead of always returning the output tensors? Is the intention to avoid populating the result vector and save some overhead?

Reply:
no, i was thinking that we might not want to define work->result() in case no result is supposed to be present. however, we are returning empty list of tensors in this case.